pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-maintenanceversion-direct/create.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: sql.cnrm.cloud.google.com/v1beta1
+kind: SQLInstance
+metadata:
+  name: sqlinstance-maintenanceversion-direct-${uniqueId}
+  annotations:
+    alpha.cnrm.cloud.google.com/reconciler: "direct"
+spec:
+  databaseVersion: POSTGRES_15
+  region: us-central1
+  # cannot set maintenanceVersion on INSERT/ CREATE
+  #maintenanceVersion: POSTGRES_15_13.R20250302.00_31
```

Comment:
Suggest changing the adaptor.Create to set `maintenanceVersion` *after* CREATE/INSERT

For declarative configurations, users do not expect to set this field in separate steps.

Reply:
That's a good point! And I'd like to understand more about this field before we add that kind of magic over the operations.

Also some behavior stuff -- what happens if we fail to set the field but we CREATED it already. Do we delete the object?

I'd expect we attempt to update the object but we have to be careful how we modify the status in the object.